src/Strategy.sol

```diff
@@ -6,36 +6,98 @@ pragma experimental ABIEncoderV2;
 import {Address} from "@openzeppelin/contracts/utils/Address.sol";
-import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
-import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {ERC20} from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import {SafeERC20, IERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
 
-// Import interfaces for many popular DeFi projects, or add your own!
-//import "../interfaces/<protocol>/<Interface>.sol";
+import {ISwap} from "./interfaces/synapse/ISwap.sol";
+import {IMasterChef} from "./interfaces/synapse/IMasterChef.sol";
+import {IUniswapV2Router02} from "./interfaces/solidly/IUniswapV2Router02.sol";
+import "../lib/forge-std/src/console.sol";
 
 contract Strategy is BaseStrategy {
     using SafeERC20 for IERC20;
     using Address for address;
 
-    // solhint-disable-next-line no-empty-blocks
-    constructor(address _vault) BaseStrategy(_vault) {
-        // You can set these parameters on deployment to whatever you want
-        // maxReportDelay = 6300;
-        // profitFactor = 100;
-        // debtThreshold = 0;
+    // tokens
+    IERC20 internal constant USDT =
+        IERC20(0x049d68029688eAbF473097a2fC38ef61633A3C7A);
+    IERC20 internal constant nUSD =
+        IERC20(0xED2a7edd7413021d440b09D654f3b87712abAB66);
+    IERC20 internal constant SYN =
+        IERC20(0xE55e19Fb4F2D85af758950957714292DAC1e25B2);
+
+    // lp tokens
+    IERC20 internal immutable syn3PoolLP;
+
+    // staking contracts
+    IMasterChef internal immutable synStakingMC;
+
+    // pools
+    ISwap internal immutable syn3PoolSwap; // Synapse Fantom Stable 3 Pool
+
+    // router
+    IUniswapV2Router02 internal immutable solidlyRouter;
+
+    //1	    0.01%
+    //5	    0.05%
+    //10	0.1%
+    //50	0.5%
+    //100	1%
+    //1000	10%
+    //10000	100%
+    uint256 public maxSlippageIn; // bips
+    uint256 public maxSlippageOut; // bips
+
+    uint256 internal constant ONE_HUNDRED_PERCENT = 10_000; // 100% bips
+
+    uint256 internal immutable pid; // Staking contract Pool ID
+    uint8 internal immutable syn3PoolUSDCTokenIndex; // Index of USDT in Synapse Fantom 3 Pool
+
+    constructor(
+        address _vault,
+        address _synStable3PoolLP,
+        address _synStable3Pool,
+        address _solidlyRouter,
+        address _synStakingMC,
+        uint256 _stakingContractPoolId,
+        uint8 _synStable3PoolUSDCTokenIndex,
+        uint256 _maxSlippageIn,
+        uint256 _maxSlippageOut
+    ) BaseStrategy(_vault) {
+        require(
+            _maxSlippageIn <= ONE_HUNDRED_PERCENT,
+            "maxSlippageIn too high"
+        );
+
+        require(
+            _maxSlippageOut <= ONE_HUNDRED_PERCENT,
+            "maxSlippageOut too high"
+        );
+        minReportDelay = 7 days;
+
+        syn3PoolLP = IERC20(_synStable3PoolLP); // FTM Mainnet: 0x2DC777ff99058a12844A33D9B1AE6c8AB4701F66
```

Comment:
can we do a check to assert this is always 18 decimals? 
if ever it's not, it will bork some hardcoded math in the scaling functions.

Reply:
Confirmed that this is always 18, and can not be changed